src/components/TodoFooter/TodoFooter.tsx

```diff
@@ -0,0 +1,93 @@
+import { useContext, useMemo } from 'react';
+import classnames from 'classnames';
+import { Status } from '../../types/Status';
+import { FilterContext } from '../../context/FilterContext';
+import { TodoContext } from '../../context/TodoContext';
+import { deleteTodo } from '../../api/todos';
+import { ErrorContext } from '../../context/ErrorContext';
+
+export const TodoFooter = () => {
+  const { todos, setTodos } = useContext(TodoContext);
+  const { selectedFilter, setSelectedFilter } = useContext(FilterContext);
+  const { setErrorMessage } = useContext(ErrorContext);
+
+  const hasCompletedTodo = useMemo(() => {
+    return todos.some(todo => todo.completed);
+  }, [todos]);
+
+  const clearCompleted = () => {
+    todos
+      .filter(({ completed }) => completed)
+      .forEach(({ id }) => {
+        deleteTodo(id)
+          .then(() => {
+            setTodos(prevState => prevState.filter(todo => todo.id !== id));
+          })
+          .catch(() => setErrorMessage('Unable to delete a todo'));
+      });
```

Comment:
There is no need to do unnecessary filtering and go through the array twice, it is enough to check the status of the todo during the traversal through the forEach.